### ISSUE TITLE
Set random seed for PCA and MDS embeddings

### ISSUE THREAD

**Issue by huddlej** [blab/pathogen-embed#3]
The `pathogen-embed` command accepts a `--random-seed` argument that is supposed to pass through to all functions that rely on random seeds. We pass this random seed to the [t-SNE](https://github.com/blab/pathogen-embed/blob/5d9c64e76a78ebbe209714046d5750a59f51746d/src/pathogen_embed/pathogen_embed.py#L263) and [UMAP](https://github.com/blab/pathogen-embed/blob/5d9c64e76a78ebbe209714046d5750a59f51746d/src/pathogen_embed/pathogen_embed.py#L273) constructors, but we don't pass the value to the [PCA](https://github.com/blab/pathogen-embed/blob/5d9c64e76a78ebbe209714046d5750a59f51746d/src/pathogen_embed/pathogen_embed.py#L247) or [MDS](https://github.com/blab/pathogen-embed/blob/5d9c64e76a78ebbe209714046d5750a59f51746d/src/pathogen_embed/pathogen_embed.py#L277-L282) constructors even though both of [PCA](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html#sklearn.decomposition.PCA) and [MDS](https://scikit-learn.org/stable/modules/generated/sklearn.manifold.MDS.html#sklearn.manifold.MDS) accept a `random_state` keyword argument.

We should update the logic for those embedding methods to include the random state from the random seed argument.
